staging/src/k8s.io/legacy-cloud-providers/gce/gcpcredential/BUILD

```diff
@@ -12,7 +12,7 @@ go_library(
         "doc.go",
         "metadata.go",
     ],
-    importpath = "k8s.io/kubernetes/pkg/credentialprovider/gcp",
+    importpath = "k8s.io/legacy-cloud-providers/gce/gcpcredential",
     deps = [
         "//pkg/credentialprovider:go_default_library",
```

Comment:
//pkg is something within the K/K private packages. We should not have a dependency from staging to a K/K private package. Please see what this dependency is and move it out as well.

Reply:
Looks like code under `//pkg/credentialprovider` is shared among multiple cloud providers and is pretty safe to be moved to staging. I would suggest to have a separated PR to move those shared files to k8s.io/cloud-provider to remove the blocker. It also sounds pretty align with the KEP

Reply:
Yeah, I'll make another PR for the common code shared between cloud providers and then rebase this PR off of that one.